editor/src/clj/editor/fs.clj

```diff
@@ -241,27 +241,34 @@
   ^File [^File directory]
   (.toFile (Files/createDirectory (.toPath directory) empty-file-attrs)))
 
+(def ^:private ^"[Ljava.nio.file.OpenOption;" overwrite-open-options (into-array OpenOption [StandardOpenOption/WRITE StandardOpenOption/CREATE StandardOpenOption/TRUNCATE_EXISTING]))
+
 (def ^:private ^"[Ljava.nio.file.LinkOption;" no-follow-link-options (into-array LinkOption [LinkOption/NOFOLLOW_LINKS]))
 
 (defn create-file!
-  "Creates a file if it does not already exist and fills it with optional UTF-8 content.
+  "Creates a file if it does not already exist and fills it with optional content.
+  If the supplied content is a byte array, write the bytes as-is to the file,
+  otherwise, write a UTF-8 string representation of the content.
   Missing parent directories are created. Returns the file."
   (^File [^File target]
    (if (not (Files/exists (.toPath target) no-follow-link-options))
      (do (create-parent-directories! target)
          (.toFile (Files/createFile (.toPath target) empty-file-attrs)))
      target))
   (^File [^File target content]
-   (create-file! target)
-   (spit target content)
-   target))
+   (let [path (.toPath target)]
+     (when (not (Files/exists path no-follow-link-options))
+       (create-parent-directories! target))
+     (if (bytes? content)
+       (Files/write path ^bytes content overwrite-open-options)
+       (spit target content))
+     target)))
```

Comment:
Note that we do not call `create-file!` from the `content`-arg variant anymore, since I want to avoid two separate file system events from creating the file and then writing to it.